.crawler/src/crawler.rs

```diff
@@ -0,0 +1,274 @@
+// Copyright (C) 2019-2022 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::{constants::*, known_network::KnownNetwork};
+use snarkos_environment::CurrentNetwork;
+use snarkos_network::Data;
+use snarkos_storage::BlockLocators;
+use snarkos_synthetic_node::{ClientMessage, SynthNode, MAXIMUM_FORK_DEPTH, MESSAGE_LENGTH_PREFIX_SIZE, MESSAGE_VERSION};
+use snarkvm::traits::Network;
+
+use pea2pea::{
+    protocols::{Disconnect, Handshake, Reading, Writing},
+    Config,
+    Node as Pea2PeaNode,
+    Pea2Pea,
+};
+use std::{convert::TryInto, io, net::SocketAddr, ops::Deref, sync::Arc, time::Duration};
+use structopt::StructOpt;
+use tokio::task;
+use tracing::*;
+
+// CLI
+#[derive(Debug, StructOpt)]
```

Comment:
I wonder if we need to use `structopt` in the crawler if the only parameter is the listening address; I think just `clap` (a dependency of `structopt`) would suffice. In any case, it's not a big deal, just a thought for later.

Reply:
Will mark this as a TODO 👍